Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -628,6 +702,25 @@ namespace Babylon
         vertexBufferData.Update(info.Env(), data, byteOffset, byteLength);
     }
 
+    // Change VS output coordinate system
+    std::string NativeEngine::ProcessShaderCoordinates(const std::string& vertexSource)
+    {
+        const auto* caps = bgfx::getCaps();
+        if (!caps->homogeneousDepth)
+        {
+            std::string patchedVertexSource;
+            const auto lastClosingCurly = vertexSource.find_last_of('}');
+            patchedVertexSource = vertexSource.substr(0, lastClosingCurly);
+            if (!caps->originBottomLeft)
+            {
+                patchedVertexSource += "gl_Position.y *= -1.;\n";
+            }
+            patchedVertexSource += "gl_Position.z = (gl_Position.z + gl_Position.w) / 2.0; }";
+            return patchedVertexSource;
+        }
+        return vertexSource;
```

Comment:
this ideally should be done with glslang, but i'm okay with a comment saying as such for now.

Reply:
I agree.